src/middleware/options/TelemetryHandlerOptions.ts

```diff
@@ -0,0 +1,86 @@
+/**
+ * -------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation.  All Rights Reserved.  Licensed under the MIT License.
+ * See License in the project root for license information.
+ * -------------------------------------------------------------------------------------------
+ */
+
+/**
+ * @module TelemetryHandlerOptions
+ */
+
+import { Context } from "../../IContext";
+import { MiddlewareControl } from "../MiddlewareControl";
+
+import { MiddlewareOptions } from "./IMiddlewareOptions";
+
+/**
+ * @enum
+ * @property {number} NONE - The hexadecimal flag value for nothing enabled
+ * @property {number} REDIRECT_HANDLER_ENABLED - The hexadecimal flag value for redirect handler enabled
+ * @property {number} RETRY_HANDLER_ENABLED - The hexadecimal flag value for retry handler enabled
+ * @property {number} AUTHENTICATION_HANDLER_ENABLED - The hexadecimal flag value for the authentication handler enabled
+ */
+export enum FeatureUsageFlag {
```

Comment:
It was my understanding that the handler pipeline should be [customizable per spec](https://github.com/microsoftgraph/msgraph-sdk-design/blob/master/middleware/middleware.md). We should add a CUSTOM_HANDLER_ENABLED flag when that work happens. Would we update IMiddleWare to require it set it at that point?

Reply:
we can figure out this from the flag value right ? let say I constructed a custom middleware which uses our Redirect handler and my own custom middlewares. in such case the feature usage flag will be set only for redirect handler. then we can conclude that the application is using custom middleware. your thoughts ?

Reply:
I just realized that I missed the createWithMiddleware in the client factory. This comment is not actionable in this PR and requires greater discussion and update to spec.

The current flag values can't be use to determine whether there is a custom pipeline or custom handlers. For example, let's say a customer creates a pipeline with our retry, redirect, telemetry, and auth handler, in addition to a custom handler. The flags would look like our default pipeline. 

What my real concern here is that as we start to see adoption of our client versions with the middleware, we are starting to see issues pop-up. One issue involved the handler order. We will see issues when customers start making changes to default handler order. We may want to investigate ways to enforce known handler order. Perhaps a middleware factory or arranger to aid customers in the creation of a pipeline - it'd have business rules for ordering default handlers. 

Another concern I have is with a custom handlers. It would be convenient to know whether a request was made with a custom handler when working with customers on issues. While customers may tell us in an issue or incident, sometimes it will be expedient to look up the logs. Knowing that a custom handler exists will help us pinpoint debugging efforts.

Reply:
Yeah, I completely agree with you. This will help a lot.